spinn_machine/machine_factory.py

```diff
@@ -128,7 +128,7 @@ def _machine_ignore(original, dead_chips, dead_links):
             router = Router(links, chip.router.emergency_routing_enabled,
                             chip.router.n_available_multicast_entries)
             chip = Chip(
-                chip.x, chip.y, chip.processors, router, chip.sdram,
+                chip.x, chip.y, chip.n_user_processors+1, router, chip.sdram,
```

Comment:
why is this not chip.n_processors? I dislike the +1 here completely.

Reply:
chip.processors is an iterator of processors

chip.n_processors although slower is saver and clearer here so changed